src/main/java/com/j256/ormlite/dao/StreamableLazyForeignCollection.java

```diff
@@ -32,39 +31,51 @@
  * {@link Stream#onClose(Runnable)} runnable to ensure that the iterator is properly closed once the
  * {@link Stream#close()} method is called.
  * </p>
- * 
+ *
  * <p>
  * WARNING: because we are currently building with a JDK before 8, this class cannot be unit tested.
  * </p>
- * 
+ *
  * @author graywatson
  */
 public class StreamableLazyForeignCollection<T, ID> extends LazyForeignCollection<T, ID> {
 
-	private static final long serialVersionUID = 1288122099601287859L;
+    private static final long serialVersionUID = 1288122099601287859L;
 
-	public StreamableLazyForeignCollection(Dao<T, ID> dao, Object parent, Object parentId, FieldType foreignFieldType,
-			String orderColumn, boolean orderAscending) {
-		super(dao, parent, parentId, foreignFieldType, orderColumn, orderAscending);
-	}
+    public StreamableLazyForeignCollection(Dao<T, ID> dao, Object parent, Object parentId, FieldType foreignFieldType, String orderColumn, boolean orderAscending) {
+        super(dao, parent, parentId, foreignFieldType, orderColumn, orderAscending);
+    }
 
-	@Override
-	public Spliterator<T> spliterator() {
-		return Spliterators.spliteratorUnknownSize(iterator(), 0);
-	}
+    @Override
+    public CloseableSpliterator<T> spliterator() {
+        final CloseableIterator<T> iterator = closeableIterator();
+        try {
+            return new CloseableSpliteratorImpl<>(iterator);
+        } catch (Error | RuntimeException e) {
+            // If something went wrong during spliterator creation we need to close the connection before re-throwing to caller.
+            iterator.closeQuietly();
+            throw e;
+        }
+    }
 
-	@Override
-	public Stream<T> stream() {
-		final CloseableIterator<T> iterator = closeableIterator();
-		/*
-		 * NOTE: we have to use a Runnable here because was want to compile via the JDK6 compiler which doesn't
-		 * understand lambdas.
-		 */
-		return StreamSupport.stream(Spliterators.spliteratorUnknownSize(iterator, 0), false).onClose(new Runnable() {
-			@Override
-			public void run() {
-				IOUtils.closeQuietly(iterator);
-			}
-		});
-	}
+    @Override
+    public Stream<T> stream() {
+        final CloseableIterator<T> iterator = closeableIterator();
```

Comment:
Is there a benefit to switching to use the `CloseableSpliterator` here?  Seems like we pay for the delegation but will never call close or closeQuietly on that object right?